Create azure.md (#17379)
DataDog/documentation: content/en/continuous_integration/pipelines/azure.md

    - link: "/continuous_integration/pipelines"
      tag: "Documentation"
      text: "Explore Pipeline Execution Results and Performance"
    - link: "/continuous_integration/pipelines/custom_commands/"
      tag: "Documentation"
      text: "Extend Pipeline Visibility by tracing individual commands"
    - link: "/continuous_integration/troubleshooting/"
      tag: "Documentation"
      text: "Troubleshooting CI"
    - link: "/continuous_integration/pipelines/custom_tags_and_metrics/"
      tag: "Documentation"
      text: "Extend Pipeline Visibility by adding custom tags and metrics"
---

{{< site-region region="gov" >}}
<div class="alert alert-warning">CI Visibility is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
{{< /site-region >}}

## Compatibility

- **Custom spans**: Configure custom spans

- **Custom pre-defined tags**: Set [custom tags][6] to all generated pipeline and job spans

- **Custom tags and metrics at runtime**: Configure [custom tags][7] and metrics at runtime

## Configure the Datadog integration

The Datadog integration for [Azure Pipelines][1] works by using [service hooks][2] to send data to Datadog.

> 1. For each project, go to **Project settings > Service hooks** in Azure DevOps and select `Create subscription`.
> 
> 2. Configure the Azure DevOps Services event.
> 
> 3. Configure what to do when the event occurs:
>     * **Service hook URL**: <code>https://webhook-intake.{{< region-param key="dd_site" >}}/api/v2/webhook/?dd-api-key=<API_KEY></code> where `<API_KEY>` is your [Datadog API key][3].

Reviewer: ```suggestion
1. For each project, go to **Project settings > Service hooks** in Azure DevOps and select `Create subscription` (green plus icon).

2. Create a new subscription with the same URL for each of the following webhook types:
    - **Run state changed**
    - **Run stage state changed**
    - **Run job state changed**
    
3. Click `Next` to continue to the next step and set the following: 
    - **URL**: <code>https://webhook-intake.{{< region-param key="dd_site" >}}/api/v2/webhook/</code>
    - **HTTP Headers**: `DD-API-KEY:<API_KEY>` where `<API_KEY>` is your [Datadog API key][3].
    - **Resource Version**: scroll down to the latest field and choose `[Latest]`.
```

---
title: Set up Tracing on an Azure Pipeline
kind: documentation
aliases:
  - /continuous_integration/setup_pipelines/azure
further_reading:
    - link: "/continuous_integration/pipelines"
      tag: "Documentation"
      text: "Explore Pipeline Execution Results and Performance"
    - link: "/continuous_integration/pipelines/custom_commands/"
      tag: "Documentation"
      text: "Extend Pipeline Visibility by tracing individual commands"
    - link: "/continuous_integration/troubleshooting/"
      tag: "Documentation"
      text: "Troubleshooting CI"
    - link: "/continuous_integration/pipelines/custom_tags_and_metrics/"
      tag: "Documentation"
      text: "Extend Pipeline Visibility by adding custom tags and metrics"
---

{{< site-region region="gov" >}}
<div class="alert alert-warning">CI Visibility is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
{{< /site-region >}}

> ## Compatibility
> 
> - **Custom spans**: Configure custom spans
> 
> - **Custom pre-defined tags**: Set [custom tags][6] to all generated pipeline and job spans
> 
> - **Custom tags and metrics at runtime**: Configure [custom tags][7] and metrics at runtime

Reviewer: ```suggestion
## Compatibility

- **Custom pre-defined tags**: Set [custom tags][6] to all generated pipeline and job spans
```
Only this will be available in the end, through the query param `tags` in the URL

`?tags=key1%3Avalue1%2C%22key2%3A+value+with+%2C+comma%22%2Ckey3%3Avalue3`

## Visualize pipeline data in Datadog

The [Pipelines][4] and [Pipeline Executions][5] pages populate with data after the workflows finish.

**Note**: The Pipelines page shows data for only the default branch of each repository.

## Further reading

{{< partial name="whats-next/whats-next.html" >}}

[1]: https://azure.microsoft.com/en-us/products/devops/pipelines
[2]: https://learn.microsoft.com/en-us/azure/devops/service-hooks/services/webhooks?view=azure-devops
[3]: https://app.datadoghq.com/organization-settings/api-keys
[4]: https://app.datadoghq.com/ci/pipelines
[5]: https://app.datadoghq.com/ci/pipeline-executions
[6]: https://docs.datadoghq.com/continuous_integration/pipelines/azure/#setting-custom-tags

> [7]: https://docs.datadoghq.com/continuous_integration/pipelines/custom_tags_and_metrics/?tab=linux

Reviewer: ```suggestion
```
